x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataStreamAndIndexLifecycleMixingTests.java

```diff
@@ -208,21 +209,15 @@ public void testIndexTemplateSwapsILMForDataStreamLifecycle() throws Exception {
             ).actionGet();
             assertThat(dataStreamLifecycleExplainResponse.getIndices().size(), is(1));
             ExplainIndexDataStreamLifecycle writeIndexDataStreamLifecycleExplain = dataStreamLifecycleExplainResponse.getIndices().get(0);
-            assertThat(writeIndexDataStreamLifecycleExplain.isManagedByLifecycle(), is(false));
+            assertThat(writeIndexDataStreamLifecycleExplain.isManagedByLifecycle(), is(true));
         });
 
-        // let's migrate this data stream to use the data stream lifecycle starting with the next generation
-        client().execute(
-            PutDataStreamLifecycleAction.INSTANCE,
-            new PutDataStreamLifecycleAction.Request(new String[] { dataStreamName }, TimeValue.timeValueDays(90))
-        );
-
         // at this point we should be able to rollover the data stream by indexing only one document (as the data stream lifecycle is
         // configured to)
         indexDocs(dataStreamName, 1);
 
-        // data stream was rolled over and has 4 indices, 2 managed by ILM, the previous write index that was UNMANAGED will now be
-        // managed by the data stream lifecycle and the new write index managed by the data stream lifecycle
+        // data stream was rolled over and has 4 indices, 2 managed by ILM, 1 managed by the data stream lifecycle and the new write index
+        // managed by the data stream lifecycle
```

Comment:
this sounds a bit strange - perhaps:
```suggestion
        // data stream was rolled over and has 4 indices, 2 managed by ILM, and the latest 2 generations managed by the data stream lifecycle 
```

Reply:
Good suggestion.